### PR TITLE
Verify that .npmignore does not contain important files 

### LINKED ISSUE #543: Feature Request: obt verify to check .npmignore
## Problem
We've had components released which have ignored critical file such as the entire demos folder due to having those paths in the `ignore` array within `bower.json`.

## Solution
Check the `ignore` array within the `bower.json` to ensure that it does not contain: `main.scss`, `main.js`, `src` or `demos`.

### PR DESCRIPTION
**Pull request by chee**
to close #543 

this is the commit that has the code: https://github.com/Financial-Times/origami-build-tools/commit/25cbffd6fa4a528fac8ffd2de2f6997f1116a8d7, the other has the test and fixtures

### CODE REVIEW

File: lib/tasks/verify-dot-npmignore.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	if (exists) {
+		const stream = createReadStream(npmignorePath);
+		const reader = readline.createInterface({
+			input: stream
+		});
+
+		let lineNumber = 0;
+		for await (const line of reader) {
+			const match = line.trim().match(/(main\.scss|main\.js|src\/?|demos\/?)/);
+			if (match) {
```
Comment (JakeChampion) [inline lib/tasks/verify-dot-npmignore.js:24]: I think this is the first `for await` I've seen in any of our projects :D

Comment (chee) [reply lib/tasks/verify-dot-npmignore.js:24]: it seems to have upset eslint: https://github.com/Financial-Times/origami-build-tools/pull/1039/checks?check_run_id=3420607722

Comment (chee) [reply lib/tasks/verify-dot-npmignore.js:24]: i've bumped our ecmaVersion to 2019

**Review by JakeChampion — APPROVED** [review]
Nice!

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v11.1.0 :tada:

**Pull request merged**